src/propslib/DelegationConfig.cpp

```diff
@@ -29,27 +29,11 @@ using namespace ABI::Windows::ApplicationModel::AppExtensions;
 #define DELEGATION_CONSOLE_KEY_NAME L"DelegationConsole"
 #define DELEGATION_TERMINAL_KEY_NAME L"DelegationTerminal"
 
-DEFINE_GUID(CLSID_SystemDelegationConsole, 0x2eaca947, 0x7f5f, 0x4cfa, 0xba, 0x87, 0x8f, 0x7f, 0xbe, 0xef, 0xbe, 0x69);
-DEFINE_GUID(CLSID_SystemDelegationTerminal, 0xe12cff52, 0xa866, 0x4c77, 0x9a, 0x90, 0xf5, 0x70, 0xa7, 0xaa, 0x2c, 0x6b);
-
 #define DELEGATION_CONSOLE_EXTENSION_NAME L"com.microsoft.windows.console.host"
 #define DELEGATION_TERMINAL_EXTENSION_NAME L"com.microsoft.windows.terminal.host"
 
-template<typename T, typename std::enable_if<std::is_base_of<DelegationConfig::DelegationBase, T>::value>::type* = nullptr>
-HRESULT _lookupCatalog(PCWSTR extensionName, std::vector<T>& vec) noexcept
+static [[nodiscard]] HRESULT _lookupCatalog(PCWSTR extensionName, std::vector<DelegationConfig::DelegationBase>& vec) noexcept
 {
-    vec.clear();
-
-    T useInbox = { 0 };
-    useInbox.clsid = { 0 };
-    // CLSID of 0 will be sentinel to say "inbox console" or something.
-    // The UI displaying this information will have to go look up appropriate strings
-    // to convey that message.
-
-    vec.push_back(useInbox);
-
-    auto coinit = wil::CoInitializeEx(COINIT_APARTMENTTHREADED);

```

Comment:
We pretty sure the propsheet initializes COM itself? Note that the propsheet can get loaded outside of the context of conhost, just by right-clicking on a `.lnk` to a commandline application.

Reply:
I just moved the initialization one call up so that we don't have to do this twice.